### DISCUSSION TITLE
Moon phase - time finder

### DISCUSSION THREAD

**Discussion by skynoob-new** [skyfielders/python-skyfield#774] [Q&A] [answered]
Hello ,
       Using below code i can find the moon phase between days, given the time,

```
from skyfield import api
from skyfield import almanac
ts = api.load.timescale()
eph = api.load('de421.bsp')
t = ts.utc(2020,7, range(1,5))
for i in t:
  phase = almanac.moon_phase(eph, i)
  print('Moon phase: {:.1f} degrees'.format(phase.degrees),i)
```

I was searching for finding the time at which a specific moonphase will occur and i got this https://rhodesmill.org/skyfield/searches.html, i am not really sure how would i combine that here to find the time at which some of the moon phase occur, ex: at what time, the moon phase would be at 20 /40/60 degrees. It seems that i cannot send a list of dates to the function `almanac.moon_phase`. Any help would be appreciated.

**Comment by skynoob-new** [comment]
Any help on this would be appreciated

**Comment by skynoob-new** [comment]
@brandon-rhodes any help on this? i dont see to find a way to get out of this ``` It seems that i cannot send a list of dates to the function almanac.moon_phase```

**Reply by brandon-rhodes** [reply]
That's unfortunate, the intention was for `moon_phase()` to support arrays of times. What error do you get when you send a `Time` that is a vector of several dates instead of just a single date?

**Reply by skynoob-new** [reply]
```
from skyfield import api
from skyfield import almanac
ts = api.load.timescale()
eph = api.load('de421.bsp')

def moonphase(t):
    phase = almanac.moon_phase(eph, t)
    return  ('Moon phase: {:.1f} degrees'.format(phase.degrees))
   
t = ts.utc(2022,7, range(1,31))

for ti, ei in zip(t, moonphase(t)):
    print((ti.utc_strftime('%b %d'), ei))
```
For the above code, i am getting the below error
```
TypeError: unsupported format string passed to numpy.ndarray.__format__
```

**Reply by brandon-rhodes** [reply]
What line of code is triggering the error?

**Reply by skynoob-new** [reply]
```
  return  ('Moon phase: {:.1f} degrees'.format(phase.degrees))
```

**Reply by skynoob-new** [reply]
```
Traceback (most recent call last):
  File "E:/myfolder/Astronomynoobexplore/moon_phase.py", line 31, in <module>
    for ti, ei in zip(t, moonphase(t)):
  File "E:/myfolder/Astronomynoobexplore/moon_phase.py", line 19, in moonphase
    return  ('Moon phase: {:.1f} degrees'.format(phase.degrees))
TypeError: unsupported format string passed to numpy.ndarray.__format__
```

**Reply by brandon-rhodes** [reply]
Thank you for the full traceback! It looks like Skyfield isn't involved in that error? It looks like it involves the `phase.degrees` object and what the code is trying to do with the Python string `.format()` method; I'd suggest printing and inspecting the `phase.degrees` object, it's probably not something you can format that way.

**Reply by skynoob-new** [reply]
i got that from here, 
```
https://rhodesmill.org/skyfield/almanac.html#phases-of-the-moon
```

**Reply by brandon-rhodes** [reply]
True, but in the example the line of code is applied to only a single phase value for a single date. Since you are giving it an array date instead, you should be getting an array back, which does not print the same way a single number prints. Again, I suggest printing and inspecting the `phase.degrees` object so that you can see more clearly what kind of value you are dealing with.

**Reply by skynoob-new** [reply]
you are right, i got it working. Thanks

**Comment by brandon-rhodes** [comment, marked as answer]
Try building a function that, instead of returning just `True` and `False`, returns an integer `0` or `1` or `2`… depending on whether the angle is between 0° and 20°, or 20° and 40°, or 40° and 60°, and so forth. If you look at Skyfield's `almanac.py`, you'll see examples like `seasons()` and `moon_phases()` that use that approach:

https://github.com/skyfielders/python-skyfield/blob/master/skyfield/almanac.py#L59

Do you think the documentation at https://rhodesmill.org/skyfield/searches.html needs to be expanded to explain this approach?
